src/main/java/org/embulk/output/GcsTransactionalFileOutput.java

```diff
@@ -57,31 +44,37 @@
         this.pathSuffix = task.getFileNameExtension();
         this.sequenceFormat = task.getSequenceFormat();
         this.contentType = task.getContentType();
-        this.maxConnectionRetry = task.getMaxConnectionRetry();
     }
 
     public void nextFile()
     {
-        closeCurrentUpload();
+        closeCurrentWriter();
         try {
-            tempFile = Exec.getTempFileSpace().createTempFile();
-            currentStream = new BufferedOutputStream(new FileOutputStream(tempFile));
-            fileIndex++;
+            String blobName = generateRemotePath(pathPrefix, sequenceFormat, taskIndex, fileIndex, pathSuffix);
+            blobId = BlobId.of(bucket, blobName);
+            BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType(contentType).build();
+            writer = client.writer(blobInfo);
         }
-        catch (IOException ex) {
-            Throwables.propagate(ex);
+        catch (Exception ex) {
+            throw new RuntimeException(ex);
         }
     }
 
     @Override
     public void add(Buffer buffer)
     {
         try {
-            logger.debug("#add called {} times for taskIndex {}", callCount, taskIndex);
-            currentStream.write(buffer.array(), buffer.offset(), buffer.limit());
-            callCount++;
+            writer.write(ByteBuffer.wrap(buffer.array(), buffer.offset(), buffer.limit()));
```

Comment:
Could we add an loading indicator for every 100MB data uploaded like below?
```
        byteCount += buffer.limit();
        if (byteCount % (100 * 1024 * 1024) < buffer.limit()) {
            logger.info("Uploaded {} megabytes", byteCount / (1024 * 1024));
        }
```

Since right now we don't have any loading indicator for how many data is processed and then with big file, it looks like the plugin is hanging